pkg/deployment/note/update.go

```diff
@@ -0,0 +1,96 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package note
+
+import (
+	"errors"
+
+	"github.com/elastic/cloud-sdk-go/pkg/api"
+	"github.com/elastic/cloud-sdk-go/pkg/client/deployments"
+	"github.com/elastic/cloud-sdk-go/pkg/models"
+	"github.com/elastic/cloud-sdk-go/pkg/util/ec"
+	multierror "github.com/hashicorp/go-multierror"
+
+	"github.com/elastic/ecctl/pkg/deployment"
+)
+
+// UpdateParams is used on Update
+type UpdateParams struct {
+	Params
+	UserID  string
+	Message string
+}
+
+// Validate confirms the parmeters are valid
+func (params UpdateParams) Validate() error {
+	var merr = new(multierror.Error)
+
+	if params.UserID == "" {
+		merr = multierror.Append(merr, errors.New(errEmptyUserID))
+	}
+
+	if params.Message == "" {
+		merr = multierror.Append(merr, errors.New(errEmptyNoteMessage))
+	}
+
+	merr = multierror.Append(merr, params.Params.Validate())
+
+	return merr.ErrorOrNil()
+}
+
+// Use different resource types when this is supported by the API.
+// For the time being, the notes endpoint only allows elasticsearch IDs.
+func (params *UpdateParams) fillDefaults() error {
+	esID, err := getElasticsearchID(deployment.GetParams{
```

Comment:
The same code appears 3 times. I wonder if we want to extract it in a single common function